### PR TITLE
Issue 1830

### PR DESCRIPTION
**Pull request by koheiw**
Fix negative indexing using a dummy named-numeric vector #1830

### CODE REVIEW

File: tests/testthat/test-corpus.R [hunk 1]
```diff
@@ -544,3 +544,27 @@ test_that("as.corpus correctly sets metadata on pre-v2 corpus", {
         substring(as.character(Sys.Date()), 1, 10)
     )
 })
+
+test_that("corpus indexing works as expected", {
+  corp <- corpus(c(d1 = "one two three", d2 = "four five six", d3 = "seven eight"))
+  
+  expect_equal(corp[[1]], "one two three")
```
Comment (koheiw) [inline tests/testthat/test-corpus.R]: Do we want the character vector to be named?
```r
expect_equal(corp[[1]], c(d1 = "one two three"))
```

Comment (kbenoit) [reply tests/testthat/test-corpus.R]: I think so. Should be same as:
```r
> texts(corp[1])
             d1 
"one two three" 
```
Sound right to you?

Comment (kbenoit) [reply tests/testthat/test-corpus.R]: I also think that if we follow the behaviour in https://github.com/quanteda/quanteda/wiki/indexing_core_objects (and I think we should), then `[[.corpus` should behave exactly as `texts.corpus()` does and therefore
- returned a named character
- allow multiple indexing.

The second is not true right now:
```r
> corp <- corpus(c(d1 = "one two three", d2 = "four five six", d3 = "seven eight"))
> corp[[c(1, 2)]]
Error in corp[[c(1, 2)]] : 
  attempt to select more than one element in vectorIndex
```

Comment (koheiw) [reply tests/testthat/test-corpus.R]: I notice that `[[.corpus` is not defined anywhere! We can allow multiple indexing but it is really unusual in R
```r
> l <- as.list(letters)
> l[[1:3]]
Error in l[[1:3]] : recursive indexing failed at level 2
> 
> letters[[1:3]]
Error in letters[[1:3]] : 
  attempt to select more than one element in vectorIndex
```

Comment (kbenoit) [reply tests/testthat/test-corpus.R]: OK, so the `[[` is consistent behaviour for a vector, which returns the unnamed contents of a single element.

```r
> c(d1 = "a", d2 = "b")[1]
 d1 
"a" 
> c(d1 = "a", d2 = "b")[[1]]
[1] "a"
```
In < v2, "[[" acts like `$.corpus` in >= v2 for returning a docvar. It's probably simpler and more consistent if we just disable `[[.corpus` and restrict `[[` to list-based objects (tokens, dictionaries).

We already cause potential confusion by hijacking (:grin:) `$`, but that's the only non-standard operator redefinition. I'd be reluctant to change `[[` as well. So: I'd say either leave it as is, or even disable it altogether.

Comment (koheiw) [reply tests/testthat/test-corpus.R]: Let's leave it as is for now.

**Pull request merged**